### PR TITLE
feat(cleanup): added automatic cleanup of stale bump branches

### PR DESCRIPTION
**Pull request by rios0rios0**
## Summary

AutoBump creates a `chore/bump-<version>` branch and opens a PR each time it runs on a repo with
unreleased changelog content. When nobody merges and the tool keeps running, stale bump branches and
their open PRs pile up — as new `[Unreleased]` entries land the computed version changes, so each run
can orphan the previous `chore/bump-<oldversion>` branch.

This adds a cleanup step: before creating the branch for the current release, every remote branch
carrying the bump prefix is deleted and the PR attached to it is closed (Azure DevOps: abandoned).

## Behaviour

- **Opt-out.** Cleanup runs unless disabled with `cleanup_stale_branches: false` or `--skip-cleanup`
  (the flag wins for a single run).
- **Only when a bump is needed.** Cleanup sits after the "is a bump needed?" check, so a PR is never
  closed without a replacement being opened for it.
- **Merged and unmerged branches alike** — a bump branch is disposable, because the one needed for the
  current release is recreated immediately afterwards.
- **Best-effort.** Every failure is logged and skipped; cleanup is housekeeping and must never abort
  the release it runs ahead of. Without a platform token the branches are still deleted and a warning
  notes the PRs were left open.
- **`bump_branch_prefix`** (default `chore/bump-`) drives branch creation *and* cleanup, so the two
  cannot drift apart and start matching different branches.

## Notes

Deleting the branch remotely is not enough on its own. `CheckBranchExists` reports a branch as
existing when it finds it either locally **or** remotely, so the surviving local reference would
convince the bumper the branch it just deleted was still there and it would skip recreating it — this
is why cleanup also calls `DeleteLocalBranch`. That was found by probing go-git's actual behaviour
rather than assuming it, and it is covered by a regression test upstream.

Depends on rios0rios0/gitforge#153 (`ClosePullRequest` on `ForgeProvider`, plus the branch helpers),
currently pinned as a pseudo-version. **This should be re-pinned to a commit on gitforge `main` once
that PR merges**, since the current pin references the PR branch.

## Testing

- New unit tests for `filterStaleBumpBranches`, `CleanupEnabled`, and `ResolveBumpBranchPrefix`
  (15 sub-tests, BDD, table of edge cases including the default branch and a custom prefix).
- `make lint` — 0 issues. `make test` — 0 failures. `make sast` — Semgrep 0 findings.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: test/domain/entitybuilders/global_config_builder.go [hunk 1]
```diff
@@ -216,6 +234,8 @@ func (b *GlobalConfigBuilder) Clone() testkit.Builder {
 		languagesConfig:        languagesConfigCopy,
 		excludeForks:           b.excludeForks,
 		excludeArchived:        b.excludeArchived,
+		cleanupStaleBranches:   b.cleanupStaleBranches,
+		bumpBranchPrefix:       b.bumpBranchPrefix,
 		changelogPath:          b.changelogPath,
```
[the remark this answers is not in the document]
Comment (rios0rios0) [reply test/domain/entitybuilders/global_config_builder.go:247]: Addressed in `fb07dfe`. `Clone` now copies the bool into a fresh pointer, so the two builders never share the address — a deep copy that hands out the same pointer is not a deep copy. Added a test that re-points the clone and asserts the original keeps its own value.

You are right that nothing mutates through the pointer today (`WithCleanupStaleBranches` takes the address of its parameter, so each call makes a new bool), but the shared address was a real trap waiting for the first test that dereferences and assigns.

**Comment by rios0rios0** [comment]
Cross-ported two fixes from the review on rios0rios0/autoupdate#215, since the same code shape exists here:

- **Failed PR close no longer deletes the branch.** Deleting anyway would strand an open PR whose source branch is gone, and because cleanup only looks at branches that still exist, no later run would see it to retry the close. `closeStalePullRequest` now reports whether the branch is safe to delete. A *missing* provider still deletes, because without a token AutoBump never opened a PR to strand in the first place.
- **Sample config shows `cleanup_stale_branches: false`.** The key is opt-out, so `true` was the one value with no effect.

`make lint` 0 issues, `make test` exit 0.
